addons/controllers/clusterbootstrap_controller.go

```diff
@@ -633,6 +640,7 @@ func (r *ClusterBootstrapReconciler) createOrPatchPackageInstallOnRemote(cluster
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      util.GeneratePackageInstallName(cluster.Name, remotePackageRefName),
 			Namespace: r.Config.SystemNamespace,
+			Labels:    map[string]string{types.ClusterNameLabel: cluster.Name, types.ClusterNamespaceLabel: cluster.Namespace},
```

Comment:
For the ```kapp-controller``` package,

Again imagine workload cluster ```wlc```, is installed under management cluster ```ns-utkg```.
kapp controller pkgi for the workload cluster is installed in the management cluster, And PackageInstall ```wlc-kapp-controller``` is installed under ```ns-utkg``` in the management cluster.  

For kapp controller, we can find workload cluster's namespace  ```ns-utkg``` from the namespace of the pkgi; But because of my previous comment for other core/additional packages, I don't think it is a good idea to define inconsistency for different packages and process each based on the existence of substring ```kapp-controller``` in their name differently.

Reply:
I get your reasoning about using the same handler to map back to cluster object for all pkgi installs but we don't expect resources to cross namespace boundary. Adding namespace here we are providing a means to be able to do that. Since kapp is special anyways I think special handler to use ownerref uid to get back to the cluster resource is fine.

Reply:
@vijaykatam Per our discussion yesterday, changed to use annotations instead of labels